pdfium/internal/subprocess/text.go

```diff
@@ -113,17 +126,32 @@ func (p *Pdfium) GetPageTextStructured(request *requests.GetPageTextStructured)
 			C.FPDFText_GetRect(textPage, C.int(i), &left, &top, &right, &bottom)
 
 			charsWritten := C.FPDFText_GetBoundedText(textPage, left, top, right, bottom, (*C.ushort)(unsafe.Pointer(&charData[0])), C.int(len(charData)))
-			pointPosition := responses.CharPosition{
-				Left:   float64(left),
-				Top:    float64(top),
-				Right:  float64(right),
-				Bottom: float64(bottom),
+			char := &responses.GetPageTextStructuredRect{
+				Text: string(removeNullTerminator(charData[0 : charsWritten*4])),
+				PointPosition: responses.CharPosition{
+					Left:   float64(left),
+					Top:    float64(top),
+					Right:  float64(right),
+					Bottom: float64(bottom),
+				},
 			}
-			resp.Rects = append(resp.Rects, &responses.GetPageTextStructuredRect{
-				Text:          string(bytes.ReplaceAll(charData[0:charsWritten*4], []byte("\x00"), []byte{})),
-				PointPosition: pointPosition,
-				PixelPosition: convertPointPositions(request.PixelPositions, pointPosition, pointToPixelRatio),
-			})
+
+			if request.CollectFontInformation {
+				// Find index of the first letter of the rect.
+				// @todo: is 5 a "valid" tolerance?
```

Comment:
maybe add a issue where this is researched

Reply:
Good one, added #23